tests/README.md

```diff
@@ -0,0 +1,194 @@
+  // - postStop - after the stop to clean up things.
+
+  // pass up eventual errors that should prohibit flushing the SUT data
+  options.cleanup = options.cleanup && opts.cleanup;
+  return rc;
+}
+
+// plugin hook. This is executed during loading of the modules.
+exports.setup = function (testFns, defaultFns, opts, fnDocs, optionsDoc, allTestPaths) {
+  // add all used testpaths, so i.e. `auto` or `find` can find the suite for single `--test` files
+  Object.assign(allTestPaths, testPaths);
+  // Register the testsuite itself
+  testFns['shell_server'] = shellServer;
+  // enable it to be ran with `./scripts/unittest all` ; if not ommit - adds the [x] in help the list.
+  defaultFns.push('shell_server');
+  // if your testsuite has CLI parameters, specify them including the default param here:
+  opts['skipAql'] = false;
+
+  // 
+  for (var attrname in functionsDocumentation) { fnDocs[attrname] = functionsDocumentation[attrname]; }
+  for (var i = 0; i < optionsDocumentation.length; i++) { optionsDoc.push(optionsDocumentation[i]); }
+};
+```
+
+### existing Javascript tests
+
+#### [permission] => tests/js/client/permissions / [server_parameters] => tests/js/client/server_parameters / [server_permissions] => client/server_permissions
+This set of testsuites is here to revalidate startup parameters of arangodb processes.
+
+The testfile persists of two parts, and the file is invoked twice: 
+
+Controlling the startup parameters can be done with the `getOptions` part. In `server_permissions` this can also be used to setup data which may not be permitted during the later run:
+```
+if (getOptions === true) {
+  return {
+    'javascript.allow-port-testing': false
+  };
+}
+```
+The later part of the test may contain regular jsunity testsuites. 
+
+If you want to revalidate a parameter that is common to arangosh & arangod, you *should* use a test in the `permission` testsuite, 
+since spawning an arangosh requires way less resources than spawning a full arangodb setup.
+
+If your testcase requires a setup with more permissions than the testcode itself, you should use `server_permissions`, since it will launch the SUT twice.
+
+You should however *prefer* `server_parameters` which will launch the SUT only once with the parameters you've provided. 
+
+You should try to use the `-[non]cluster` filename filters to also control whether your test is ran on cluster too to reduce computing power.
+
+#### [dump*] to test dump and restore procedures
+The dump family of testsuites lives in `js/client/modules/@arangodb/testsuites/dump.js`
+and orchestrates a sequence:
+- launching a SUT
+- create test data within it
+- dump this test data
+- execute some manual sanity checks on the dump files
+- restore it to a fresh SUT
+- check data
+
+Since many create & check functions should be common to *all* suites, the default behavior is to enable them, and only blacklist them out if unsupported.
```

Comment:
```suggestion
Since many create & check functions should be common to *all* suites, the default behavior is to enable them, and only exclude them if unsupported.
```